pkg/module/zip_read_closer.go

```diff
@@ -0,0 +1,50 @@
+package module
+
+import (
+	"io"
+	"os"
+
+	"github.com/gomods/athens/pkg/errors"
+	"github.com/spf13/afero"
+)
+
+type zipReadCloser struct {
+	zip    io.ReadCloser
+	fs     afero.Fs
+	goPath string
+}
+
+// Close closes the zip file handle and clears up disk space used by the underlying disk ref
+// It is the caller's responsibility to call this method to free up utilized disk space
+func (rc *zipReadCloser) Close() error {
+	rc.zip.Close()
+	return ClearFiles(rc.fs, rc.goPath)
+}
+
+func (rc *zipReadCloser) Read(p []byte) (n int, err error) {
+	return rc.zip.Read(p)
+}
+
+// ClearFiles deletes all data from the given fs at path root
+// This function must be called when zip is closed to cleanup the entire GOPATH created by the diskref
+func ClearFiles(fs afero.Fs, root string) error {
```

Comment:
no need to be exposed

Reply:
it's used here as well:
https://github.com/gomods/athens/blob/7ee25d59d6930be4e963c332c949b411a6915ebd/pkg/download/upstream_lister.go#L61